### ISSUE TITLE
ScrollView snapToInterval rounding issue

### ISSUE THREAD

**Issue by jkk** [facebook/react-native#9172]
I'm having two issues with the ScrollView and snapToInterval:
1. Stopping an in-progress scroll by tapping in place often snaps to an unexpected interval offset.
2. Panning a small amount _always_ results in snapping to the next or previous interval offset.

Here's what the behavior in `master` looks like for my app (notice the struggle to get the green card to stay in center position):

![scrollview-snaptointerval-floor](https://cloud.githubusercontent.com/assets/5173/17342157/e101201a-58c5-11e6-8f81-0246b6b48eb6.gif)

Here's what ideal behavior (for my app) looks like, after I patched the `scrollViewWillEndDragging` method in `React/Views/RCTScrollView.m` to use `round` instead of `floor`, and to remove the directional check:

![scrollview-snaptointerval-round](https://cloud.githubusercontent.com/assets/5173/17342166/e6eb9a78-58c5-11e6-9b8b-5acb825b2132.gif)

In the modified version, it's easy to stop the ScrollView on the intended card, and it only scrolls to the next/previous snap interval when you pan past a threshold.

I could create a patch for this, but arguably the current behavior isn't wrong, just undesired for my use case. "Fixing" it could be considered a breaking change. On the other hand, the current behavior feels pretty unnatural to me. Perhaps an additional prop could be added to control this? `snapBias`? Or `snapThreshold`?

In the meantime, I'm using a patched version for my app.

EDIT: here's the patch I applied to `React/Views/RCTScrollView.m`:

``` patch
@@ -713,8 +713,7 @@
     }

     // Pick snap point based on direction and proximity
-    NSInteger snapIndex = floor((targetContentOffsetAlongAxis + alignmentOffset) / snapToIntervalF);
-    snapIndex = (translationAlongAxis < 0) ? snapIndex + 1 : snapIndex;
+    CGFloat snapIndex = round((targetContentOffsetAlongAxis + alignmentOffset) / snapToIntervalF);
     CGFloat newTargetContentOffset = ( snapIndex * snapToIntervalF ) - alignmentOffset;

     // Set new targetContentOffset

```

**Comment by anttimo** [comment]
Hi,

I have a pr open #6906 that fixes the issue where ScrollView snaps back if you touch the screen while it's scrolling forward. It's bit more complex than your fix but it doesn't change the snapping behaviour otherwise. After seeing this ticket I decided to update the fix and examples again and will try to ping somebody so it gets reviewed. If you don't mind can you also check it out and say what do you think about it? :)

**Comment by jkk** [comment]
@anttimo Thanks for the heads up! I didn't notice your pull request. Your change fixes my first issue, which I do think is a bug. My second issue -- pan threshold for scrolling prev/next -- is more like a new feature, and would probably require a new prop. If there's interest from maintainers, I can create a separate pull request for that.

**Comment by anttimo** [comment]
Np.! I agree that having a new prop for snap threshold would not be a bad
idea.

**Comment by anttimo** [comment]
As an update, looks like my fix is coming in v0.36.0 https://github.com/facebook/react-native/commit/72e203bf9572f2dbd6e836f35db3afdfdba3d6c4#diff-973dcd0e3fbfcc5e35485bb6404a2fd2

**Comment by lacker** [comment]
Closing since anttimo's fix is merged.
